framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -188,6 +190,38 @@ public int runTests() throws Exception {
 
       LOG.info("> RUNNING TESTS (ITERATION " + i + ")..");
       Collections.shuffle(tests, new Random());
+      if (cmdParam.repeatRun == true) {
+        String queryLogFilename = drillTestDataDir + "/queryLog.log";
+        File file = new File(queryLogFilename);
+        if (file.exists() && (file.length() > 0)) {
+          try (BufferedReader reader = new BufferedReader(new FileReader(new File(
+                                       queryLogFilename)))) {
+            List<DrillTest> tests_copy = Lists.newArrayList(tests);
+            tests.clear();
+            String line = "";
+            while ((line = reader.readLine()) != null) {
+              for (DrillTest test : tests_copy) {
+                if (test.getInputFile().equals(line)) {
+                  boolean result = tests_copy.remove(test);
+                  if (result) {
+                    tests.add(test);
+                  }
+                  break;
+                }
+              }
+            }
+            reader.close();
+          }
+        }
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File(
```

Comment:
This can be optimized, i.e. we probably don't need to write to the file again if it exists already.